airbyte-integrations/connectors/source-google-sheets/pyproject.toml

```diff
@@ -0,0 +1,33 @@
+[build-system]
+requires = [ "poetry-core>=1.0.0",]
+build-backend = "poetry.core.masonry.api"
+
+[tool.poetry]
+version = "0.3.15"
+name = "source-google-sheets"
+description = "Source implementation for Google Sheets."
+authors = [ "Airbyte <contact@airbyte.io>",]
+license = "MIT"
+readme = "README.md"
+documentation = "https://docs.airbyte.com/integrations/sources/google-sheets"
+homepage = "https://airbyte.com"
+repository = "https://github.com/airbytehq/airbyte"
+
+[tool.poetry.dependencies]
+python = "^3.9"
+requests = "==2.31.0"
```

Comment:
Why hard pin? I get it if we're just moving from `setup.py` as it was, but figured in theory we want to unpin most dependencies to major get at least patch / minor version upgrades for free?

Reply:
My take is that `Poetry.lock` gives us the deterministic build, but the actual dependency list should be as flexible as possible. If it was previously hard-pinned in setup.py, it might be because we were trying to do the lock thing with the deps list. I would try and err on the side of unpinning. 

Perhaps, though, this should be another step after this all is done. For example, note that the CDK is pinned. I don't think we should unpin it right now, that might cause issues ;)

Reply:
@natikgadzhi the idea there is to force this new connector version to have the same dependency version as the previous one. Even if the previous version was not pinning a version I want to pin main deps in this newest version to guarantee no regression.
These pins can be relaxed without issues in upcoming PRs if needed.